core/services/im_here/MavlinkNMEA.py

```diff
@@ -0,0 +1,86 @@
+from enum import IntEnum
+from typing import Optional
+
+import pynmea2
+from pydantic import BaseModel
+
+from exceptions import UnsupportedSentenceType
+
+
+class GPS_FIX_TYPE(IntEnum):
+    GPS_FIX_TYPE_NO_GPS = 0
+    GPS_FIX_TYPE_NO_FIX = 1
+    GPS_FIX_TYPE_2D_FIX = 2
+    GPS_FIX_TYPE_3D_FIX = 3
+    GPS_FIX_TYPE_DGPS = 4
+    GPS_FIX_TYPE_RTK_FLOAT = 5
+    GPS_FIX_TYPE_RTK_FIXED = 6
+    GPS_FIX_TYPE_STATIC = 7
+    GPS_FIX_TYPE_PPP = 8
+
+
+class GPS_INPUT_IGNORE_FLAG(IntEnum):
+    GPS_INPUT_IGNORE_FLAG_ALT = 1
+    GPS_INPUT_IGNORE_FLAG_HDOP = 2
+    GPS_INPUT_IGNORE_FLAG_VDOP = 4
+    GPS_INPUT_IGNORE_FLAG_VEL_HORIZ = 8
+    GPS_INPUT_IGNORE_FLAG_VEL_VERT = 16
+    GPS_INPUT_IGNORE_FLAG_SPEED_ACCURACY = 32
+    GPS_INPUT_IGNORE_FLAG_HORIZONTAL_ACCURACY = 64
+    GPS_INPUT_IGNORE_FLAG_VERTICAL_ACCURACY = 128
+
+
+class Mavlink2RestBitEnum(BaseModel):
+    bits: int
+
+
+nmea_ignore_flags = (
+    GPS_INPUT_IGNORE_FLAG.GPS_INPUT_IGNORE_FLAG_SPEED_ACCURACY
+    + GPS_INPUT_IGNORE_FLAG.GPS_INPUT_IGNORE_FLAG_VEL_VERT
+    + GPS_INPUT_IGNORE_FLAG.GPS_INPUT_IGNORE_FLAG_VEL_HORIZ
+)
+
+
+class MavlinkGeoPackage(BaseModel):
```

Comment:
docstring would be interesting. is this based on the GPS_INPUT message?

Reply:
It is the GPS_INPUT indeed! Will change the name and add a docstring.